qiita_db/artifact.py

```diff
@@ -0,0 +1,598 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from __future__ import division
+from future.utils import viewitems
+from datetime import datetime
+
+from .base import QiitaObject
+from .sql_connection import TRN
+from .exceptions import (QiitaDBArtifactCreationError,
+                         QiitaDBArtifactDeletionError,
+                         QiitaDBOperationNotPermittedError)
+from .util import (convert_to_id, insert_filepaths,
+                   move_filepaths_to_upload_folder, retrieve_filepaths)
+from .software import Parameters, Command
+from .study import Study
+from .metadata_template import PrepTemplate
+
+
+class Artifact(QiitaObject):
+    r"""A piece of data that different commands can be applied on
+
+    Attributes
+    ----------
+    timestamp
+    processing_parameters
+    visibility
+    artifact_type
+    can_be_submitted_to_ebi
+    can_be_submitted_to_vamps
+    is_submitted_to_vamps
+    filepaths
+    parents
+    prep_template
+    ebi_run_accession
+    study
+
+    Methods
+    -------
+    create
+    delete
+
+    See Also
+    --------
+    qiita_db.QiitaObject
+    """
+    _table = "artifact"
+
+    @classmethod
+    def create(cls, filepaths, artifact_type, timestamp=None,
```

Comment:
I'm not sure what you mean in here. I think the description in the parameters already explain when they are optional. The only one that doesn't have that kind of explanation is timestamp. The only reason that I'm actually exposing that parameter through the API is in case that we need to upload something manually through the CLI.